scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamFunctions.scala

```diff
@@ -22,4 +25,31 @@ object BeamFunctions {
     def processElement(c: DoFn[A, B]#ProcessContext): Unit =
       c.output(f(c.element()))
   }
+
+  case class MapSideAggregator[K, V](size: Int, semigroup: Semigroup[V]) extends DoFn[KV[K, V], KV[K, V]] {
+    var cache: SummingCache[K, V] = _
+    @StartBundle
+    def startBundle(): Unit = {
+      cache = new SummingCache[K, V](size)(semigroup)
+    }
+    @ProcessElement
+    def processElement(c: DoFn[KV[K, V], KV[K, V]]#ProcessContext): Unit = {
+      val key = c.element().getKey
+      val value = c.element().getValue
+      val evicted = cache.put(Map((key, value)))
+      evicted match {
+        case Some(m) => m.foreach(x => c.output(KV.of(x._1, x._2)))
```

Comment:
nit: using a while loop and `m.iterator` is faster (no closure):
```scala
val mit = m.iterator
while (mit.hasNext) {
  val x = mit.next()
  c.output(KV.of(x._1, x._2))
}
```
A bit of a pain to write things like that out, but avoiding closures can be measurable wins in some of the internals.